Dockerfile.build

```diff
@@ -2,13 +2,16 @@
 # --- stage 1 ---
 FROM registry.ci.openshift.org/openshift/release:golang-1.16 as builder
 
+ENV GO111MODULE=on
+
 WORKDIR /build
 
 RUN echo "noroot:x:10001:10001:noroot:/:/sbin/nologin" | tee -a /etc/passwd
 
 # Cache optimization
-COPY go.mod go.sum ./
-RUN go mod download
+COPY go.mod go.sum Makefile ./
+COPY vendor ./vendor
```

Comment:
```suggestion
```
can be removed as the `make vendor` instruction in the next line brings back the vendor files anyway :)

Reply:
I thought it would speed up if we copy instead of redownloading?

As a side note, I saw they removed the `vendor/` dir in the `addon-operator` all together. Should we do the same? I feel since go modules this is not really required, and mostly annoying (especially in commit history). WDYT?

Reply:
I am always up for getting rid of vendor/ , messes up a lot of PRs. Although I can think of only a very few (1-2) benefits of vendoring, it seems like an overkill to me. So yeah! let's purge it \o/

Regarding addon-operator, some context: we added some submodules underneath it and for making that compliant and not end up vendoring our own repo, we purged vendoring.